packages/core/src/typeschema/crawler.ts

```diff
@@ -18,36 +18,80 @@ export interface ResourceVisitor {
   ) => void;
 }
 
+export interface AsyncResourceVisitor {
+  onEnterObject?: (path: string, value: TypedValueWithPath, schema: InternalTypeSchema) => Promise<void>;
+  onExitObject?: (path: string, value: TypedValueWithPath, schema: InternalTypeSchema) => Promise<void>;
+  onEnterResource?: (path: string, value: TypedValueWithPath, schema: InternalTypeSchema) => Promise<void>;
+  onExitResource?: (path: string, value: TypedValueWithPath, schema: InternalTypeSchema) => Promise<void>;
+  visitPropertyAsync: (
+    parent: TypedValueWithPath,
+    key: string,
+    path: string,
+    value: TypedValueWithPath | TypedValueWithPath[],
+    schema: InternalTypeSchema
+  ) => Promise<void>;
+}
+
+function isSchema(obj: any): obj is InternalTypeSchema {
```

Comment:
Super minor: the TS team seems to be pushing for preferring `unknown` over `any` for a long list of safety and correctness reasons.  We still have have a lot of `any` floating around.  I'm kinda treating it as a personal goal to be better about it for new code.  This might be a good one for a team brown bag discussion.